app/models/comment.rb

```diff
@@ -38,8 +38,34 @@ def create_subscription
   def notify(action)
     case action.to_s
     when 'create'
+      subscribers_by_mention = subscribe_mentions()
+      create_notifications(action: :mention, recipients: subscribers_by_mention)
+
       subscribers = commentable.subscriptions.where.not(user: user).map(&:user)
-      create_notifications(action: :create, recipients: subscribers)
+      create_notifications(action: :create, recipients: subscribers - subscribers_by_mention)
     end
   end
+
+  def mentions
+    users = []
+    HTML::Pipeline::MentionFilter.mentioned_logins_in(content) do |match, login, is_mentioned|
+      if (mentioned_user = User.find_by_email(login))
+        users << mentioned_user
+      end
+    end
+
+    users
+  end
+
+  private
+  
+  def subscribe_mentions
+    subscribers_by_mention = []
+    mentions.each do |mention|
+      Subscription.subscribe(user: mention, to: commentable)
+      subscribers_by_mention << mention
+    end
+
+    subscribers_by_mention
```

Comment:
there is something weird here, `subscribers_by_mention` is just a duplicate of the return value of `#mentions`.

Why not avoid this and have `create` L40 first subscribe and then call `#mentions` again to pass that to `create_notifications`. Should `#mentions` be memoized? Possibly dependent on the value of `:content` having changed?